## People Analytics: Employee Insights Dashboard

## Motivo do projeto -
    Desenvolver uma solução de análise de dados voltada para Recursos Humanos, utilizando indicadores de colaboradores para transformar informações brutas em insights estratégicos sobre retenção, distribuição da força de trabalho, remuneração e perfil dos funcionários.

Exploração e preparação dos dados -

In [3]:
#Importando a biblioteca e carregando o dataset
import pandas as pd

df = pd.read_csv("../WA_Fn-UseC_-HR-Employee-Attrition.csv")
df.head()


,Age,Attrition,BusinessTravel,DailyRate,Department,DistanceFromHome,Education,EducationField,EmployeeCount,EmployeeNumber,...,RelationshipSatisfaction,StandardHours,StockOptionLevel,TotalWorkingYears,TrainingTimesLastYear,WorkLifeBalance,YearsAtCompany,YearsInCurrentRole,YearsSinceLastPromotion,YearsWithCurrManager
0,41,Yes,Travel_Rarely,1102,Sales,1,2,Life Sciences,1,1,...,1,80,0,8,0,1,6,4,0,5
1,49,No,Travel_Frequently,279,Research & Development,8,1,Life Sciences,1,2,...,4,80,1,10,3,3,10,7,1,7
2,37,Yes,Travel_Rarely,1373,Research & Development,2,2,Other,1,4,...,2,80,0,7,3,3,0,0,0,0
3,33,No,Travel_Frequently,1392,Research & Development,3,4,Life Sciences,1,5,...,3,80,0,8,3,3,8,7,3,0
4,27,No,Travel_Rarely,591,Research & Development,2,1,Medical,1,7,...,4,80,1,6,3,3,2,2,2,2


In [4]:
df.describe(include='object')

C:\Users\adhel\AppData\Local\Temp\ipykernel_14932\87514550.py:1: Pandas4Warning: For backward compatibility, 'str' dtypes are included by select_dtypes when 'object' dtype is specified. This behavior is deprecated and will be removed in a future version. Explicitly pass 'str' to `include` to select them, or to `exclude` to remove them and silence this warning.
See https://pandas.pydata.org/docs/user_guide/migration-3-strings.html#string-migration-select-dtypes for details on how to write code that works with pandas 2 and 3.
  df.describe(include='object')


,Attrition,BusinessTravel,Department,EducationField,Gender,JobRole,MaritalStatus,Over18,OverTime
count,1470,1470,1470,1470,1470,1470,1470,1470,1470
unique,2,3,3,6,2,9,3,1,2
top,No,Travel_Rarely,Research & Development,Life Sciences,Male,Sales Executive,Married,Y,No
freq,1233,1043,961,606,882,326,673,1470,1054


## Perguntas de negócio -

### 1. Total de funcionários ativos:

In [5]:
# Filtrando os funcionários ativos
df_ativos = df[df["Attrition"] == "No"].copy()
df_ativos["Attrition"].value_counts()

Attrition
No    1233
Name: count, dtype: int64

Total de 1233 funcionários ativos.

### 2. Porcentagem de Turnover:

In [6]:
turnover = (df["Attrition"].value_counts(normalize=True)["Yes"] * 100).round(2)

turnover

np.float64(16.12)

O total de 16,12% de Turnover

O cálculo do turnover permite medir a proporção de funcionários que deixaram a empresa, sendo um indicador importante para avaliar retenção de talentos e possíveis impactos na gestão de pessoas.

## 3) Idade média para funcionários ativos:

In [7]:
df_ativos["Age"].mean().round(2)

np.float64(37.56)

A média para a idade de funcionários é de 37 anos.

## 4) Salário médio:

In [8]:
df_ativos["MonthlyIncome"].mean().round(2)

np.float64(6832.74)

O salário médio dos funcionários é de 6.832,74.

## 5) Total de funcionários por departamento:

In [9]:
df_ativos["Department"].value_counts()

Department
Research & Development    828
Sales                     354
Human Resources            51
Name: count, dtype: int64

Pesquisa e Desenvolvimento: 828.
Vendas: 354.
RH: 51.

## 6. Percentual de funcionários por gênero:

In [10]:
df_ativos["Gender"].value_counts(normalize=True).mul(100).round(2)

Gender
Male      59.37
Female    40.63
Name: proportion, dtype: float64

Homens: 59,37%
Mulheres: 40,63%

## 7) Tempo médio de empresa por departamento:

In [11]:
df_ativos.groupby("Department")["YearsAtCompany"].mean().round(2)

Department
Human Resources           7.96
Research & Development    7.17
Sales                     7.75
Name: YearsAtCompany, dtype: float64

RH & Vendas possuem funcionários com quase 8 anos de empresa, enquanto Pesquisa e Desenvolvimento posue funcionários com 7 anos. 

## 8. Total de funcionários por cargo:

In [ ]:
df_ativos["JobRole"].value_counts().sort_values(ascending=False)

JobRole
Sales Executive              269
Research Scientist           245
Laboratory Technician        197
Manufacturing Director       135
Healthcare Representative    122
Manager                       97
Research Director             78
Sales Representative          50
Human Resources               40
Name: count, dtype: int64

A distribuição dos funcionários por cargo permite identificar quais funções concentram a maior parte da força de trabalho, auxiliando no planejamento de equipes, dimensionamento de recursos e análise da estrutura organizacional.

## 9) Faixa salarial por cargo:

In [ ]:
df_ativos.groupby("JobRole")["MonthlyIncome"].mean().round(2).sort_values(ascending=False)

JobRole
Manager                      17201.48
Research Director            15947.35
Healthcare Representative     7453.56
Manufacturing Director        7289.93
Sales Executive               6804.62
Human Resources               4391.75
Laboratory Technician         3337.22
Research Scientist            3328.12
Sales Representative          2798.44
Name: MonthlyIncome, dtype: float64

A análise salarial por cargo permite compreender a distribuição da remuneração dentro da empresa, identificando diferenças entre funções e auxiliando em decisões relacionadas à estrutura salarial e planejamento de carreira.

## Visualização de Dados:
    Para criar um dashboard interativo no Power BI, será feito antes a separação de colunas necessárias para que as análises feitas aqui estejam expostas da maneira correta, eliminando qualquer outra que não faça sentido com o projeto.

In [ ]:
colunas_dashboard = [
    "Attrition",
    "Department",
    "Gender",
    "Age",
    "MonthlyIncome",
    "JobRole",
    "YearsAtCompany"
]

df_dashboard = df[colunas_dashboard].copy()

In [ ]:
df_dashboard.head()

,Attrition,Department,Gender,Age,MonthlyIncome,JobRole,YearsAtCompany
0,Yes,Sales,Female,41,5993,Sales Executive,6
1,No,Research & Development,Male,49,5130,Research Scientist,10
2,Yes,Research & Development,Male,37,2090,Laboratory Technician,0
3,No,Research & Development,Female,33,2909,Research Scientist,8
4,No,Research & Development,Male,27,3468,Laboratory Technician,2


In [ ]:
df_dashboard.to_csv("dados_people_analytics.csv", index=False)